### DISCUSSION TITLE
print preview enable

### DISCUSSION THREAD

**Discussion by wnunezc** [cefsharp/CefSharp#3514] [Q&A]
I'm looking for an improvement to the print preview and follow-up for this post::

https://bitbucket.org/chromiumembedded/cef/issues/123/add-support-for-print-preview#comment-55952406

![image](https://user-images.githubusercontent.com/1207306/115289028-c36dba00-a117-11eb-8f01-8184fab12b17.png)

but in the client cef it says that you need to use:  --enable-print-preview

does this option exist in cefsharp?
Can it be enabled programmatically?

i am getting this print preview, is to poor:

![image](https://user-images.githubusercontent.com/1207306/115289455-324b1300-a118-11eb-8358-d8199bc943ef.png)

**Comment by amaitland** [comment]
> does this option exist in cefsharp?

Yes

> Can it be enabled programmatically?

For examples of setting command line args see https://github.com/cefsharp/CefSharp/wiki/General-Usage#cefsettings-and-browsersettings you can also use the helper method http://cefsharp.github.io/api/89.0.x/html/M_CefSharp_CefSettingsBase_EnablePrintPreview.htm

> i am getting this print preview, is to poor:

You'll have to raise any quality issues on https://magpcss.org/ceforum/index.php 
There are no configuration options provided by `CEF` that I'm aware of.

Further questions about print preview would be best asked on `ceforum` (post a link back here for reference to others).

**Reply by wnunezc** [reply]
Thank you very much: I have a code that looks like the following C #:

```C#
public void InitializeChromium(string Url)
        {
            CefSettings settings = new CefSettings();
            Cefinit = Cef.Initialize(settings);
            BrowserControl = new ChromiumWebBrowser(Url)
            {
                Name = "WebBrowser",
                MenuHandler = new SearchContextMenuHandler(this),
                DownloadHandler = new DownloadHandler(download_img)
            };
            Controls.Add(BrowserControl);
        }
```
what and where should I add the method: `EnablePrintPreview()`
I don't see any implementation example, just a link to a page that says that something exists without any kind of detail.

http://cefsharp.github.io/api/89.0.x/html/M_CefSharp_CefSettingsBase_EnablePrintPreview.htm

I'm looking for it to look like this or similar:

https://user-images.githubusercontent.com/1207306/115289028-c36dba00-a117-11eb-8f01-8184fab12b17.png
https://bitbucket.org/chromiumembedded/cef/issues/123/add-support-for-print-preview#comment-55952406

**Reply by amaitland** [reply]
A simple example would look like:

```c#
var settings = new CefSettings();
settings.EnablePrintPreview();

var success = Cef.Initialize(settings);
```

Only `CefSharp.WinForms` and `CefSharp.Wpf.HwndHost` support print preview. This is a `CEF` limitation.

**Comment by baby1silly** [comment]
is you resolve this questions?

**Reply by amaitland** [reply]
What is your question exactly?

**Comment by JoshuaD10** [comment]
@amaitland Is there a way to programmatically set the default page size to A4?

**Reply by amaitland** [reply]
Not that I'm aware of. Try asking on https://www.magpcss.org/ceforum/index.php for a second opinion.
